go-sdk/common/format.go

```diff
@@ -50,7 +50,7 @@ const (
 )
 
 type Format struct {
-	Unit          string `json:"unit" yaml:"unit"`
+	Unit          string `json:"unit,omitempty" yaml:"unit,omitempty"`
 	DecimalPlaces int    `json:"decimalPlaces,omitempty" yaml:"decimalPlaces,omitempty"`
 	ShortValues   bool   `json:"shortValues,omitempty" yaml:"shortValues,omitempty"`
 }
```

Comment:
Btw in the same file you should also add this at the beginning of `validate()`:

```go
if f.Unit == nil {
    return nil
}
```

Otherwise your initial issue wont be solved. Btw maybe worth updating the unit tests to match this case then.